### PR TITLE
Docs for v1

### PR DESCRIPTION
**Pull request by voliva**
This is an initial proposal for docs.

The current readme is targeted to explain specifically how is `redux-views` different from `reselect` and how does it effectively solves the specific problem on shared selectors.

I wanted to make a readme that's more like a standalone selector library, without comparing it to others (given that most likely this will become reselect@5.0).

I also added two sections for migrating from reselect@4.0, and re-reselect, as it's one of the most popular ways of dealing with the problem that `redux-views` solves by default.

There's still work to be done, and it's just my initial idea of the structure. I'm pretty happy to change everything, as we haven't really discussed any of this.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,206 +1,193 @@
...
+    getRunningCars,
+    getPassengers,
+  ],
+  (cars, passengers) => ...
+);
+```
 
-const getPropIdA = createIdSelector(({idA}) => idA)
-const getPropIdB = createIdSelector(({idB}) => idB)
+That's the only breaking change for the simple usage. `redux-views` passes all the tests from `reselect@4.0`. There's a codemod in the works that will allow to replace all of these automatically.
```
Comment (voliva) [inline README.md:121]: I think I lied here... Now I see it's not the only breaking change, there's also `defaultMemoize` and `createSelectorCreator` that were removed. We'll have to document how to migrate those as well....

File: README.md [hunk 2]
```diff
@@ -1,206 +1,295 @@
...
-with shared selectors (almost) transparent for the user.
+## Installation
 
-The API of Redux-Views is almost identical to the API of `reselect`. Basically, it keeps `reselect`'s 2 high-level functions:
-- `createSelector` has the exact same signature.
-- `createStructureSelector` is identical to the one provided by `reselect` except that it only accepts the first argument.
-
-On top of those 2 functions, Redux-Views adds a new function: `createIdSelector`, which helps the library to identify those selectors that just return ids.
+```sh
+yarn add redux-views
```
Comment (josepot) [inline README.md:8]: nitpick: I would prefer `npm install --save redux-views`

File: README.md [hunk 3]
```diff
@@ -1,206 +1,295 @@
...
 
-getUser.recomputations() // => 4
+assert(activeCars1 === activeCars2);
+assert(getRunningCars.recomputations() === 1);
 ```
 
-## Redux-Views solution
+## Parametric selectors
+
+More often than not, we need to build selectors that take parameters, usually to identify a specific instance. Imagine we want to build a selector that returns the passengers of a specific car. If we write the selector in a plain function, we can get an idea of what we need, but it won't memoize the results:
```
Comment (josepot) [inline README.md:38]: I would make some changes to this paragraph... For instance:

> Sometimes we need to build selectors that take parameters, usually to identify a specific instance. 

> Consider the following selector which returns the passengers of a specific car:

File: README.md (hunk 1 above)
Comment (josepot) [reply README.md:121]: > We'll have to document how to migrate those as well....

I don't know about that... I'm of the opinion that those were implementation details and I would much rather to ask the community how and why they were using those. I hope that in most cases we will be able to either suggest another alternative or that we will find a use-case to improve the API of this library.

In short: I wouldn't worry too much about that.

Maybe we could rephrase it to:

> This is the main breaking for this usage.

**Comment by josepot** [comment]
If I have to be 100% honest:

**Comment by josepot** [comment]
Thanks a lot for this! I ended-up cherry-picking these commits, because of the conflict issues.
